src/userservice/tests/test_db.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for db module
+"""
+
+import unittest
+from unittest.mock import patch
+
+from sqlalchemy.exc import IntegrityError
+
+from userservice.db import UserDb
+from userservice.tests.constants import EXAMPLE_USER
+
+
+class TestDb(unittest.TestCase):
+    """
+    Test cases for db module
+    """
+
+    def setUp(self):
+        """Init db and create table before each test"""
+        # init SQLAlchemy with sqllite in mem
+        self.db = UserDb('sqlite:///:memory:')
+        # create users table in mem
+        self.db.users_table.create(self.db.engine)
+
+    def test_add_user_foo(self):
+        """test if a user can be added"""
+        user = EXAMPLE_USER.copy()
+        # create a user with username foo
+        user['username'] = 'foo'
+        user['accountid'] = '1'
+        # add user to db
+        self.db.add_user(user)
+
+    def test_add_same_user(self):
+        """test if one user can be added twice"""
+        user = EXAMPLE_USER.copy()
+        # create a user with username bar
+        user['username'] = 'bar'
+        user['accountid'] = '2'
+        # add bar_user to db
+        self.db.add_user(user)
+        # try to add same user again
+        self.assertRaises(IntegrityError, self.db.add_user, user)
+
+    def test_get_user(self):
+        """test getting a user"""
+        user = EXAMPLE_USER.copy()
+        # create a user with username baz
+        user['username'] = 'baz'
+        user['accountid'] = '3'
+        # add baz_user to db
+        self.db.add_user(user)
+        # get baz_user from db
+        db_user = self.db.get_user(user['username'])
+        # assert both user objects are equal
+        self.assertEqual(user, db_user)
+
+    def test_get_non_existent_user(self):
```

Comment:
(nit) Consider using more descriptive names for your test methods
(small, for every unit test method)

A common best practice template I see is `functionname_situation_expectedresult`.  Obviously, long method names are expected.
For example: `test_get_nonexistent_user_returns_none`

Resources:
[Naming standards for unit tests](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html)
[Python testing style guidelines](https://jrsmith3.github.io/python-testing-style-guidelines.html)
[Testing Your Code](https://docs.python-guide.org/writing/tests/)
etc, etc...

Reply:
great suggestion, helpful to see what failed without digging into test code.